wormchain/x/wormhole/keeper/query_plugins.go

```diff
@@ -0,0 +1,78 @@
+package keeper
+
+import (
+	"encoding/json"
+
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+)
+
+func NewCustomQueryHandler(keeper Keeper) *wasmkeeper.QueryPlugins {
+	return &wasmkeeper.QueryPlugins{
+		Custom: WormholeQuerier(keeper),
+	}
+}
+
+type WormholeQuery struct {
+	VerifyQuorum    *verifyQuorumParams    `json:"verify_quorum,omitempty"`
+	VerifySignature *verifySignatureParams `json:"verify_signature,omitempty"`
+	CalculateQuorum *calculateQuorumParams `json:"calculate_quorum,omitempty"`
+}
+
+type verifyQuorumParams struct {
+	Data             []byte           `json:"data"`
+	GuardianSetIndex uint32           `json:"guardian_set_index"`
+	Signatures       []*vaa.Signature `json:"signatures"`
+}
+
+type verifySignatureParams struct {
+	Data             []byte         `json:"data"`
+	GuardianSetIndex uint32         `json:"guardian_set_index"`
+	Signature        *vaa.Signature `json:"signature"`
```

Comment:
I think this is currently causing an encoding mismatch.  This type has a `SignatureData` field, which implements `MarshalJSON` but not `UnmarshalJSON`.  The `MarshalJSON` encodes the signature as a hex string but since there is no `UnmarshalJSON` it falls back to expecting an array of bytes.

Meanwhile, the rust bindings use `cosmwasm_std::Binary` for the same field, which encodes the signature into a base64 string.

As part of #1932 I changed this field to always encode as an array of bytes but given the existing `MarshalJSON` method for `vaa.SignatureData` that does something different we should define our own `Signature` type here for encoding purposes and then provide a conversion function to convert to `vaa.Signature`.